File: src/CircleAI.Wearable.Biosignals/BiosignalKind.cs
// BiosignalKind.cs
//
// Canonical taxonomy of biosignals ingested by Circle AI's wearable layer.
// Integer values are stable across language ports — do not renumber.

namespace CircleAI.Wearable.Biosignals;

/// <summary>
/// Canonical kinds of biosignal samples Circle AI consumes from wearables.
/// </summary>
public enum BiosignalKind
{
    /// <summary>Heart rate, beats per minute.</summary>
    HeartRate = 0,

    /// <summary>Heart rate variability, RMSSD in milliseconds.</summary>
    HeartRateVariability = 1,

    /// <summary>Peripheral oxygen saturation, percent (0-100).</summary>
    OxygenSaturation = 2,

    /// <summary>Accelerometer magnitude, m/s^2.</summary>
    Accelerometer = 3,

    /// <summary>Body temperature, degrees Celsius.</summary>
    BodyTemperature = 4,

    /// <summary>Sleep stage encoded as a float: 0=awake, 1=light, 2=deep, 3=REM.</summary>
    SleepStage = 5,

    /// <summary>Step count (cumulative or delta — see <see cref="BiosignalSample.IsCumulative"/>).</summary>
    Steps = 6,

    /// <summary>Galvanic skin response, microsiemens.</summary>
    GalvanicSkinResponse = 7,

    /// <summary>Catch-all for vendor-specific or future signals.</summary>
    Unknown = 8,
}

File: python/src/circle_ai/wearable_biosignals/biosignal_kind.py
# biosignal_kind.py
#
# Port of CircleAI.Wearable.Biosignals BiosignalKind.cs (C# — the EXACT spec).
#
# Canonical taxonomy of biosignals ingested by Circle AI's wearable layer.
# Integer values are stable across language ports — do not renumber.

from __future__ import annotations

from enum import IntEnum


class BiosignalKind(IntEnum):
    """Canonical kinds of biosignal samples Circle AI consumes from wearables.

    Mirrors ``CircleAI.Wearable.Biosignals.BiosignalKind``.
    """

    #: Heart rate, beats per minute.
    HEART_RATE = 0
    #: Heart rate variability, RMSSD in milliseconds.
    HEART_RATE_VARIABILITY = 1
    #: Peripheral oxygen saturation, percent (0-100).
    OXYGEN_SATURATION = 2
    #: Accelerometer magnitude, m/s^2.
    ACCELEROMETER = 3
    #: Body temperature, degrees Celsius.
    BODY_TEMPERATURE = 4
    #: Sleep stage encoded as a float: 0=awake, 1=light, 2=deep, 3=REM.
    SLEEP_STAGE = 5
    #: Step count (cumulative or delta — see BiosignalSample.is_cumulative).
    STEPS = 6
    #: Galvanic skin response, microsiemens.
    GALVANIC_SKIN_RESPONSE = 7
    #: Catch-all for vendor-specific or future signals.
    UNKNOWN = 8
